catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java

```diff
@@ -580,6 +584,25 @@ private String getBindingTypeToActivate(List<ServiceBindingType> bindingTypes) {
     return bindingTypeToActivate;
   }
 
+  private Dictionary<String, Object> copyConfigProperties(
+      Dictionary<String, Object> properties, String factoryPid) {
+    Dictionary<String, Object> copiedProperties = new Hashtable<>();
+    ObjectClassDefinition objectClassDefinition = getObjectClassDefinition(factoryPid);
+    if (objectClassDefinition == null) {
+      LOGGER.debug(
+          "ObjectClassDefinition not found for factoryPid: {}. Unable to copy properties.",
+          factoryPid);
+      return copiedProperties;
+    }
+    Stream.of(objectClassDefinition)
+        .map(ocd -> ocd.getAttributeDefinitions(ObjectClassDefinition.ALL))
+        .flatMap(Arrays::stream)
+        .map(AttributeDefinition::getID)
+        .filter(id -> properties.get(id) != null)
```

Comment:
✏️ Not a fan of accessing `properties` twice. Once to check if it has an entry and once to retrieve it.
How about creating a simple private method like:

```
private void copyIfDefined(String id, Dictionary<String, Object> source, Dictionary<String, Object> destination) {
  final Object value = source.get(id);

  if (value != null) {
    destination.put(id, value);
  }
}
```

and then simply doing `.forEach(id -> ccopyIfDefinedpy(properties, copiedProperties, id))`